renderdoc/driver/gl/gl_counters.cpp

```diff
@@ -206,7 +206,6 @@ struct GLCounterContext
 };
 
 GLenum glCounters[] = {
-    eGL_NONE,                                      // Undefined!!
```

Comment:
I don't understand the motivation behind this change, it seems to me like it breaks things.

The `GPUCounter` enum starts with `GPUCounter::EventGPUDuration == 1`, so that a value of 0 serves as a useful uninitialised/invalid value. The `glCounters` array then matches up to each index then, as in the comments, `GL_NONE` is provided corresponding to the `Undefined!!` value to pad the indices, but it should never be actually submitted to `glBeginQuery` or `glEndQuery`. Removing it means that each counter is off-by-one.

When you were testing did you get correct results? If I try your changes I get invalid values when I run a simple timer query for example, and stepping through I see that when `GPUCounter::EventGPUDuration` is requested, `GL_VERTICES_SUBMITTED_ARB` is actually submitted and read back so the values are nonsensical. It also means that submitting `GPUCounter::CSInvocations` reads off the end of the array and submits an invalid counter.

Reply:
Then maybe the loop in FillTimers need to be modified : 
```
for(int32_t q = uint32_t(GPUCounter::Count) - 1; q >= 0; q--)
```
That'll reach 0 which tries to create a query of type GL_NONE.

Reply:
Actually the problem must be somewhere else for GPUCounter = 0 to end up in the list of counters to capture.
I'll keep digging.
Thanks!

Reply:
Can you clarify - that loop calls `glBeginQuery` it's not the one that creates queries with `glGenQueries`.

The `glBeginQuery` loop should be fine because it checks that `queries->obj[q]` is valid, and the `[0]`th element should always be `0`. When calling `glGenQueries` we only initialise entries in the `obj[]` array corresponding to requested `GPUCounter` values, which start at `1`.

Reply:
If `GPUCounter == 0` is ending up in the list of counters that's a problem, as it should never be reported by `GLReplay::EnumerateCounters`.

Are you testing with only the basic counters, or are you also testing with your additional PR of intel counters? Perhaps there's a bug in that code causing `GPUCounter == 0` to be reported there. It would be worth adding some sanity-checking code to make sure 0 is never included, to catch the bug closer to the source.

Reply:
Ok I should probably drop those 2 first patches.
It seems that the first one introduced an issue that I tried to fix in the second one.
Works fine without those 2.

It probably all came from new counter considered as generic (because !AMD) though they were Intel ones.

Thanks again.